x-pack/filebeat/module/panw/panos/ingest/pipeline.yml

```diff
@@ -1,616 +1,530 @@
-     allow_duplicates: false
-     value:
-       - '{{destination.nat.ip}}'
-
-# Geolocation for source.
- - geoip:
-     if: 'ctx?.source?.ip != null'
-     field: source.ip
-     target_field: source.geo
-
-# Geolocation for destination.
- - geoip:
-     if: 'ctx?.destination?.ip != null'
-     field: destination.ip
-     target_field: destination.geo
-
-# IP Autonomous System (AS) Lookup
- - geoip:
-     database_file: GeoLite2-ASN.mmdb
-     field: source.ip
-     target_field: source.as
-     properties:
-       - asn
-       - organization_name
-     ignore_missing: true
- - geoip:
-     database_file: GeoLite2-ASN.mmdb
-     field: destination.ip
-     target_field: destination.as
-     properties:
-       - asn
-       - organization_name
-     ignore_missing: true
- - rename:
-     field: source.as.asn
-     target_field: source.as.number
-     ignore_missing: true
- - rename:
-     field: source.as.organization_name
-     target_field: source.as.organization.name
-     ignore_missing: true
- - rename:
-     field: destination.as.asn
-     target_field: destination.as.number
-     ignore_missing: true
- - rename:
-     field: destination.as.organization_name
-     target_field: destination.as.organization.name
-     ignore_missing: true
-
-# Set source|destination.geo.name from panw's srcloc|dstloc
- - rename:
-     if: 'ctx.source?.geo?.name == null'
-     field: _temp_.srcloc
-     target_field: source.geo.name
-     ignore_missing: true
- - rename:
-     if: 'ctx.destination?.geo?.name == null'
-     field: _temp_.dstloc
-     target_field: destination.geo.name
-     ignore_missing: true
-
-# Append NAT community_id to network.community_id
- - append:
-     if: 'ctx?.panw?.panos?.network?.nat?.community_id != null'
-     field: network.community_id
-     allow_duplicates: false
-     value:
+        }
+
+## TRAFFIC
+  - pipeline:
+      if: ctx?.panw?.panos?.type == "TRAFFIC"
+      name: '{< IngestPipeline "traffic" >}'
+
+## THREAT
+  - pipeline:
+      if: ctx?.panw?.panos?.type == "THREAT"
+      name: '{< IngestPipeline "threat" >}'
+
+## GLOBAL PROTECT
+  - pipeline:
+      if: ctx?.panw?.panos?.type == "GLOBALPROTECT"
+      name: '{< IngestPipeline "globalprotect" >}'
+
+## USER ID
+  - pipeline:
+      if: ctx?.panw?.panos?.type == "USERID"
+      name: '{< IngestPipeline "userid" >}'
+
+  - append:
+      field: event.type
+      allow_duplicates: false
+      value: allowed
+      if: "ctx?.panw?.panos?.action != null && ['alert', 'allow', 'continue'].contains(ctx.panw.panos.action)"
+  - append:
+      field: event.type
+      allow_duplicates: false
+      value: denied
+      if: "ctx?.panw?.panos?.action != null && ['deny', 'drop', 'reset-client', 'reset-server', 'reset-both', 'block-url', 'block-ip', 'random-drop', 'sinkhole', 'block'].contains(ctx.panw.panos.action)"
+  - set:
+      field: event.outcome
+      value: failure
+      if: "ctx?.event?.type != null && ctx?.event?.type.contains('denied')"
```

Comment:
This is causing a lot of events in the traffic logs to be marked as `failure`. The way I see it, the event types set by the processor above should always be `success`, as in, the firewall was successful in denying a connection, dropping traffic or blocking an URL.

Reply:
Copy that, @leehinman and I had talked about that in another PR. I'll update that.

Reply:
done